node/core/pvf/worker/src/common.rs

```diff
@@ -152,6 +152,9 @@ fn kill_parent_node_in_emergency() {
 }
 
 /// Functionality related to threads spawned by the workers.
+///
+/// The motivation for this module is to coordinate worker threads without using async Rust. This
+/// lets us pull in less dependencies, making the worker binaries smaller and easier to secure.
```

Comment:
Threads should be coordinated without async because there're no yield points and every thread runs blocking code.

We don't use async deps not because of dependencies, but because "by design".